crates/pico/src/lib.rs

```diff
@@ -1,3 +1,53 @@
-pub mod container;
-pub mod memo;
-pub mod storage;
+use std::{
+    any::Any,
+    hash::{DefaultHasher, Hash, Hasher},
+};
+
+use crate::index::Index;
+
+mod arena;
+mod database;
+mod dependency;
+mod derived_node;
+mod dyn_eq;
+mod epoch;
+mod generation;
+mod index;
+mod memo;
+mod source;
+mod u64_types;
+
+pub use database::*;
+pub use derived_node::*;
+pub use memo::*;
+pub use source::*;
+pub use u64_types::*;
+
+pub fn intern_param<T: Hash + Clone + 'static>(db: &Database, param: T) -> ParamId {
```

Comment:
not blocking, but I'll move this (and any other fns I find that are only used by the macro) into a sub module